lib/GLPI/Agent/SNMP/MibSupport/Aruba.pm

```diff
@@ -83,6 +85,50 @@ sub getModel {
     return "AP $model";
 }
 
+sub run {
+    my ($self) = @_;
+
+    my $device = $self->device
+        or return;
+
+    # Get list of device ports (e.g. radioX_ssid_idY)
+    my $ports = $device->{PORTS}->{PORT};
+
+    # Equivalent to "show ap bss-table" Aruba IAP CLI output command:
+
+    # Get list of SSID
+    my $aiWlanESSIDValues = $self->walk(aiWlanESSID) || {};
+    # Get list of Radios (e.g. radioX_ssid_idY etc.)
+    my $aiWlanMACAddressValues = $self->walk(aiWlanMACAddress) || {};
+    # The list of Radios is co-related to the list of SSIDs
+    # $aiWlanMACAddressValues->{0} = radio0_ssid_id0
+    # $aiWlanESSIDValues->{0} = <SSID>
+
+    foreach my $index (keys(%$aiWlanMACAddressValues)) {
+        # Get WLAN BSSID (e.g. XX:XX:XX:XX:XX:XX)
+        my $wlanMacAddress = getCanonicalMacAddress($aiWlanMACAddressValues->{$index});
+
+        foreach my $port (keys(%$ports)) {
+            my $ifMacAddress = $device->{PORTS}->{PORT}->{$port}->{MAC};
+            next unless defined($ifMacAddress) && $ifMacAddress eq $wlanMacAddress;
+
+            my $ifDescr = $device->{PORTS}->{PORT}->{$port}->{IFDESCR};
+
+            # Defines the port alias with the name of the radio (e.g. radioX_ssid_idY)
+            $device->{PORTS}->{PORT}->{$port}->{IFALIAS} = $ifDescr;
+            # Replaces the radio port name with its respective <SSID>
+            $device->{PORTS}->{PORT}->{$port}->{IFNAME} = getCanonicalString($aiWlanESSIDValues->{$index});
+
+            # radio0 and radio1 are the network interfaces for the 5GHz and 2.4GHz radios respectively
+            if ($ifDescr =~ m/^radio(0)/) {
+                $device->{PORTS}->{PORT}->{$port}->{IFNAME} .= " (5GHz)";
+            } elsif ($ifDescr =~ m/^radio(1)/) {
+                $device->{PORTS}->{PORT}->{$port}->{IFNAME} .= " (2.4GHz)";
+            }
```

Comment:
Here, you have no guaranty IFNAME will be defined, so you should take some care. Parenthesis are not required in last matching checks, and also I think you forgot to add a `last` statement to leave the port search:
```suggestion
            # Replaces the radio port name with its respective <SSID>
            my $ifName = getCanonicalString($aiWlanESSIDValues->{$index});
            unless (empty($ifName)) {
                # radio0 and radio1 are the network interfaces for the 5GHz and 2.4GHz radios respectively
                if ($ifDescr =~ m/^radio0/) {
                    $ifName .= " (5GHz)";
                } elsif ($ifDescr =~ m/^radio1/) {
                    $ifName .= " (2.4GHz)";
                }

                $device->{PORTS}->{PORT}->{$port}->{IFNAME} = $ifName;
            }
            last;
```